### ISSUE TITLE
Pyarrow GLIBC_2.14 not found

### ISSUE THREAD

**Issue by harplife** [apache/arrow#6012]
Hello!

I apologize in advance if I'm barking at the wrong tree.
I wasn't sure whether I should be raising this issue on Ray, Arrow, or on Modin,
as it concerns all of them.

The gist of the problem is this:
1. Tried to import Modin using Ray as its dependency
2. I get ImportError: /lib64/libc.so.6: version 'GLIBC_2.14' not found
3. Well, there really isn't GLIBC_2.14 on the current system. There's GLIBC_2.12 tho

Possible solutions I have found so far:
1. Update the GLIBC from version 2.12 to 2.14 (which will break the system)
2. Install another GLIBC and make a symbolic link (on /lib64/libc.so.6) pointing to it (which broke the system)
3. Install another GLIBC and edit Pyarrow to import that instead?
4. Install another GLIBC and use Patchelf to direct python interpreter to it (haven't tried it yet)
5. Install higher version of the current os that comes with GLIBC 2.14 (I'd rather not tho)

I'd very much prefer Option 3, if I could just figure out a way to tell Pyarrow to look for the other GLIBC that's installed on the system. I'd appreciate any help I can get.

Current settings:
OS : Centos 6.10
Python : 3.6
ENV : conda
ray : 0.7.3
modin : 0.6.3
pyarrow : 0.15.1

**Comment by wesm** [comment]
Sounds like you have a mixed up toolchain. How did you install each package? Last I checked Ray was vendoring pyarrow entirely and building it with a different C++ toolchain, so that would certainly cause these issues. I suggest “uninstalling” pyarrow and using the one that’s inside Ray and also to ask the Ray developers to depend on released versions of pyarrow if that fixes the problem. 

If there is some other issue can you write to our mailing lists or open a JIRA issue? Thanks

**Comment by harplife** [comment]
I didn't really install each package separately, as I just relied on Conda to install necessary packages with compatible versions (supposedly what Conda package manager is good for).
I just used 'conda install modin[ray]'.
As far as "uninstalling" goes, pip can search for the package just fine but it won't uninstall - pyarrow that's being used is inside Ray afterall.

I'll take this to Ray as per your suggestion.

Thanks.
